### DISCUSSION TITLE
Need some help with Image Magick running on Windows 10

### DISCUSSION THREAD

**Discussion by xender3** [ImageMagick/ImageMagick#2017] [General]
I am setting up a windows .bat file to automatically crop, resize and animate a series of GOES satellite images that I receive from the satellite receiver I built.
This is my current batch file that works: I would Like to be able to change the section in italics to [read](https://gbapps.net/rc-yowhatsapp-apk/) all subdirectories in F:\Satellite Imagery\goes16\fd\NOMAP\ (all subdirectories) without having to type the folder path in like 2020-03-28, 2020-03-30, 2020-03-31, etc. with a 'for loop' but have no idea on how to accomplish this, not from a command line but rather in this batch file, any thoughts?
___________________

REM 4/4/2020

magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-27\GOES16_*.jpg" +repage

magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-28\GOES16_*.jpg" +repage

magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-29\GOES16_*.jpg" +repage

magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-30\GOES16_*.jpg" +repage

magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-31\GOES16_*.jpg" +repage

magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "F:\Satellite Imagery\goes16\fd\NOMAP\2020-04-01\GOES16_*.jpg" +repage

magick mogrify -path "F:/CONUS/Watermarked/" -format jpg -gravity northwest -draw "image over 30,30 0,0 'F:/Satellite Imagery/GOESFULLDISKWHITE.png'" "F:/CONUS/GOES*.jpg"

magick mogrify -path "F:/CONUSResized/" -filter Triangle -define filter:support=2 -thumbnail 1000 -unsharp 0.25x0.25+8+0.065 -dither None -posterize 136 -quality 82 -define jpeg:fancy-upsampling=off -define png:compression-filter=5 -define png:compression-level=9 -define png:compression-strategy=1 -define png:exclude-chunk=all -interlace none -colorspace sRGB -strip "F:/CONUS/*.jpg"

magick convert -loop 0 -delay 10 "F:\CONUS\CONUSResized\GOES16_*" "F:/CONUS/CONUSfinal/CONUSout.gif"
pause
________________________

Here is my folder structure
F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-27
F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-28
F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-29
F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-30
F:\Satellite Imagery\goes16\fd\NOMAP\2020-03-31
F:\Satellite Imagery\goes16\fd\NOMAP\2020-04-01
etc.

**Comment by snibgo** [comment]
> ... to read all subdirectories in F:\Satellite Imagery\goes16\fd\NOMAP\ (all subdirectories) without having to type the folder path in like 2020-03-28, 2020-03-30, 2020-03-31, etc. with a 'for loop' but have no idea on how to accomplish this, ...

Have you tried "help for"?

The "/D" option does what I think you need. For example, Windows BAT syntax:

for /D %%D in ("F:\Satellite Imagery\goes16\fd\NOMAP\") do magick mogrify -path ../CONUS -format jpg -crop 2753x1769+510+269 "%%D\GOES16_*.jpg" +repage

However, your "+repage" should probably be before the final argument, which should be the filename.

Incidentally, I suggest you don't use directories (or files) with spaces in their names. Spaces just make life difficult.
